search/src/dockit-search.ts

```diff
@@ -0,0 +1,260 @@
+import { LitElement, css, html } from 'lit';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+
+const beforeHitMatch = 20;
+const afterHitMatch = beforeHitMatch;
+const markChar = '\0';
+
+export class SearchBox extends LitElement {
+  search = null;
+  hits = null;
+
+  constructor() {
+    super();
+    addEventListener('keydown', (e) => {
+      if ((e.ctrlKey || e.metaKey) && e.key === 'k') {
+        e.preventDefault();
+        this.updateHits();
+      } else if (e.key === 'Escape') this.hideHits();
+    });
+  }
+
+  render() {
+    return html`<div id="search-box">
+      <form action="" role="search" novalidate="">
+        <label for="search-input" aria-label="Search">
+          <input
+            aria-autocomplete="list"
+            id="search-input"
+            autocomplete="off"
+            autocorrect="off"
+            autocapitalize="off"
+            spellcheck="false"
+            maxlength="64"
+            type="search"
+            enterkeyhint="go"
+            placeholder=${this.placeHolder}
+            @input=${(e: { target: { value: string } }) =>
+              this.updateHits(e.target.value)}
+            @focus=${(e: { target: { value: string } }) =>
+              this.updateHits(e.target.value)}
+          />
+        </label>
+      </form>
+      <ul id="search-hits">
+        ${this.hits === null
+          ? html`<li><i>Type for searching...</i></li>`
```

Comment:
`aria-polite` might be necessary here and for `No match found`, did you test in VoiceOver what happens when you search, use keyboard to navigate up/down, when there are no search results? did you intend to support all these scenarios for screen reader?

Reply:
it was a port from what we have.
I don't care about screenreader I am not doing a newspaper search or google stuff.

I tried to use the combox from lion, but it is really a mess to understand and configure. There is almost no doc and example are not that helpful. So I decided to go straightforward with simple HTML tags. and basic ul/li and input. I think it's accessible enough for my usage. We can for sure improve it later.

Reply:
It's already ready working for keyboard navigation thx for basic html